bin/all_sky_search/pycbc_bin_trigger_rates_dq

```diff
@@ -0,0 +1,164 @@
+    g = h5.File(filename, 'r')
+    g_logl = g[ifo + '/dq_val'][:]
+    g_times = g[ifo + '/times'][:]
+    g.close()
+    dq_logl = np.concatenate((dq_logl,g_logl))
+    dq_times = np.concatenate((dq_times,g_times))
+del g_logl
+del g_times
+
+# todo: make this configurable
+percent_bin = 0.5
+n_bins = int(100./percent_bin)
+percentiles = np.linspace(0,100,n_bins+1)
+bin_times = np.zeros(n_bins)
+dq_percentiles = np.percentile(dq_logl,percentiles)[1:]
+
+    
+# seconds bin tells what bin each second ends up
+seconds_bin = np.array([n_bins - \
+                        len(dq_percentiles[dq_percentiles >= dq_ll]) \
+                        for dq_ll in dq_logl]).astype('int')
+del dq_percentiles
+
+# bin times tells how much time ends up in each bin
+bin_times = np.array([len(seconds_bin[seconds_bin==(i)]) \
+                     for i in range(n_bins)]).astype('float')
+full_time = float(len(seconds_bin))
+del dq_logl
+
+# create a dict to look up dq percentile at any time
+dq_percentiles_time = dict(zip(dq_times, seconds_bin*percent_bin/100))
+del dq_times
+               
+if args.prune_number>0:
+    for bin_name in locs_names:
+        logging.info('Processing bin %s...'%bin_name)
+        bin_locs = locs_dict[bin_name]
+        trig_times_bin = trig_times[np.isin(trig_ids, bin_locs)] 
+        trig_stats_bin = stat[np.isin(trig_ids, bin_locs)]
+    
+        for j in range(args.prune_number):        
+            max_stat_arg = np.argmax(trig_stats_bin)
+            remove = np.nonzero(abs(trig_times_bin[max_stat_arg] - trig_times)
+                               < args.prune_window)[0]
+            remove_inbin = np.nonzero(abs(trig_times_bin[max_stat_arg] \
+                               - trig_times_bin) < args.prune_window)[0]
+            stat[remove] = 0
+            trig_stats_bin[remove_inbin] = 0
+    keep = np.nonzero(stat)[0]
+    trig_times_int = trig_times_int[keep]
+    trig_ids = trig_ids[keep]
+    del stat
+    del keep
+
+del trig_times
+               
+f = h5.File(args.output_file,'w')        
+for bin_name in locs_names:
+    bin_locs = locs_dict[bin_name]
+    trig_times_bin = trig_times_int[np.isin(trig_ids, bin_locs)] 
+    trig_percentile = np.array([dq_percentiles_time[t] \
+                                for t in trig_times_bin])
+    logging.info('Processing %d triggers...'%len(trig_percentile))
+    
+    (counts, bins) = np.histogram(trig_percentile, bins = (percentiles)/100)
+    counts = counts.astype('float')
+    rates = np.zeros(len(bin_times))
+    times_nz = (bin_times > 0)
+    rates[times_nz] = counts[times_nz]/bin_times[times_nz]
+    mean_rate = len(trig_percentile) / full_time
+    if mean_rate > 0.:
+        rates = rates / mean_rate
+    
+    logging.info('Writing rates to output file %s...'%args.output_file)
+    grp = f.create_group(bin_name)
+    grp['rates']=rates
+    grp['locs']=locs_dict[bin_name]
+ 
+f.attrs['names'] = locs_names
+f.close()
```

Comment:
Finish by logging "Done" here, as you did for other scripts. When looking at logs to debug problems, I find it very helpful when a process reports explicitly that it has completed successfully.

Reply:
Done